ggml-quants.c

```diff
@@ -1273,7 +1273,12 @@ static float make_qkx2_quants(int n, int nmax, const float * restrict x, const f
     float max = x[0];
     float sum_w = weights[0];
     float sum_x = sum_w * x[0];
-    for (int i = 1; i < n; ++i) {
+    #if defined(__APPLE__) && defined(__clang_major__) && __clang_major__ >= 15
+        // use 'volatile' to prevent unroll and work around a bug in Apple clang 15.x.x with -O3 flag
+        for (volatile int i = 1; i < n; ++i) {
+    #else
+        for (int i = 1; i < n; ++i) {
+    #endif
```

Comment:
```suggestion
#if defined(__APPLE__) && defined(__clang_major__) && __clang_major__ >= 15
    // use 'volatile' to prevent unroll and work around a bug in Apple clang 15.x.x with -O3 flag
    for (volatile int i = 1; i < n; ++i) {
#else
    for (int i = 1; i < n; ++i) {
#endif
```